daemon/src/monitor.rs

```diff
@@ -515,6 +570,58 @@ where
 
         Ok(())
     }
+
+    async fn rebroadcast_transactions(
+        &mut self,
+        monitor: &(impl MessageChannel<TryBroadcastTransaction> + 'static),
+    ) -> Result<()> {
+        let mut conn = self.db.acquire().await?;
+        for id in db::load_all_cfd_ids(&mut conn).await? {
+            let (_, events) = db::load_cfd(id, &mut conn).await?;
+
+            let Cfd {
+                monitor_lock_finality,
+                monitor_commit_finality,
+                cet,
+                commit_tx,
+                lock_tx,
+                ..
+            } = events.into_iter().fold(Cfd::default(), Cfd::apply);
+
+            if !monitor_commit_finality {
```

Comment:
This should not be necessary. The `apply` logic should set `commit_tx` to `Some` if you should broadcast and `None` if you shouldn't.

Reply:
I have removed the finality checks and I rebroadcast if the txns are `Some`. I think we need to change the aggregate. Sounds like this should be an enum with three variants rather than 3 options.